Console/Command/NodesValidatorCommand.php

```diff
@@ -0,0 +1,203 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Snowdog\Menu\Console\Command;
+
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\App\Area;
+use Magento\Framework\App\State;
+use Magento\Framework\Exception\CouldNotDeleteException;
+use Magento\Framework\Exception\LocalizedException;
+use Snowdog\Menu\Api\Data\MenuInterface;
+use Snowdog\Menu\Api\Data\NodeInterface;
+use Snowdog\Menu\Api\MenuRepositoryInterface;
+use Snowdog\Menu\Api\NodeRepositoryInterface;
+use Snowdog\Menu\Model\ImportExport\Processor\Import\Node\Validator;
+use Snowdog\Menu\Model\ImportExport\Processor\Import\Validator\ValidationAggregateError;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputArgument;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Question\ConfirmationQuestion;
+
+/**
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
+ */
+class NodesValidatorCommand extends Command
+{
+    private MenuRepositoryInterface $menuRepository;
+    private NodeRepositoryInterface $nodeRepository;
+    private SearchCriteriaBuilder $searchCriteriaBuilder;
+    private Validator $validator;
+    private ValidationAggregateError $validationAggregateError;
+    private State $state;
+
+    public function __construct(
+        MenuRepositoryInterface $menuRepository,
+        NodeRepositoryInterface $nodeRepository,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        Validator $validator,
+        ValidationAggregateError $validationAggregateError,
+        State $state,
+        string $name = null
+    ) {
+        $this->menuRepository = $menuRepository;
+        $this->nodeRepository = $nodeRepository;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->validator = $validator;
+        $this->validationAggregateError = $validationAggregateError;
+        $this->state = $state;
+
+        parent::__construct($name);
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function configure()
+    {
+        $description = 'It will loop all menu nodes and return the IDs and description of invalid menu nodes.'
+            . PHP_EOL
+            . ' Additionally, it will ask to remove the detected invalid node if needed.';
+
+        $this->setName('snowdog-menu:validate-nodes')
+            ->setDescription($description);
+
+        parent::configure();
+    }
+
+    public function execute(InputInterface $input, OutputInterface $output)
```

Comment:
What do you think about wrapping whole logic in a separate class?

Reply:
I think we can, but also I think it's not needed, its logic is not super complex and also useful outputs are shown while command is being executed (which could make more complex to be abstracted)

Anyways if you have a clear idea of how to improve it, please let me know!